lib/pbench/server/api/resources/upload_api.py

```diff
@@ -124,20 +121,32 @@ def put(self, controller):
         md5_full_path = Path(path, f"{filename}.md5")
         bytes_received = 0
 
-        # TODO: Need real user from PUT!
-
         # Create a tracking dataset object; it'll begin in UPLOADING state
         try:
             dataset = Dataset(
                 owner=username, controller=controller, path=tar_full_path, md5=md5sum
             )
             dataset.add()
+        except DatasetDuplicate:
+            self.logger.info("Dataset already exists {}", filename)
+            abort(
+                200, message="Dataset already exists",
+            )
```

Comment:
Hmm. An abort with an error message and success status is a bit weird. You're raising an HttpException which gets caught by Flask to cause the early return.

I think ultimately this does what we want, but I think that just doing a `return` here with the response we want would be cleaner and less likely to be messed up by something like a generic error handler later.

Reply:
Yes I also think return would be more appropriate choice here